### ISSUE TITLE
Query / Invoke fails with a 5 node peer network (security enabled and pbft batch mode)

### ISSUE THREAD

**Issue by ratnakar-asara** [hyperledger-archives/fabric#1503]
## Description

After an initial deployment of Chaincode Exampl02 on 5 peer network, Invoke/Query fails to execute
## Describe How to Reproduce
1. Setup a 5 peer network with security enabled
2. Deploy chaincode Example02 with some initial values
3. One of the PEER goes down, after repeated Invokes and/or Queries on the chaincode (could see the issue in first two attempts)
- PEER0 Exited after an Invoke
- A gist of logs from PEER4:

```
[[31m02:41:11.466 [ledger] sendProducerBlockEvent -> ERRO 032[0m Error unmarshalling deployment transaction for block event: proto: protos.ChaincodeDeploymentSpec: wiretype end group for non-group
02:41:17.232 [consensus/obcpbft] execDoneSync -> INFO 033[0m Replica 4 finished execution 1, trying next
02:41:27.384 [consensus/obcpbft] executeOne -> INFO 034[0m Replica 4 executing/committing request for view=0/seqNo=2 and digest 8wQEiRJGpDSwwus1rLCB9NrM/FsMktZigW4pwppDLJlM06iYEOOVc97000aYSvoVdgxeAkzsoN6CR/BlebCOnA==
2016/05/18 02:41:37 transport: http2Client.notifyError got notified that the client transport was broken EOF.
[31m02:41:39.834 [peer] handleChat -> ERRO 035[0m Error during Chat, stopping handler: rpc error: code = 13 desc = "transport is closing"
2016/05/18 02:41:39 grpc: ClientConn.transportMonitor exits due to: grpc: timed out trying to connect
[31m02:41:39.876 [peer] start -> ERRO 036[0m Error sending DISC_GET_PEERS during handler discovery tick: Error Sending message through ChatStream: rpc error: code = 13 desc = "transport: write tcp 172.17.0.7:54511->172.17.0.1:30001: use of closed network connection"
[31m02:41:39.877 [peer] chatWithPeer -> ERRO 037[0m Ending chat with peer address=172.17.0.1:30001 due to error:  Error during Chat, stopping handler: rpc error: code = 13 desc = "transport is closing"
2016/05/18 02:41:41 transport: http2Client.notifyError got notified that the client transport was broken read tcp 172.17.0.7:54593->172.17.0.1:30001: read: connection reset by peer.
2016/05/18 02:41:41 transport: http2Client.notifyError got notified that the client transport was broken read tcp 172.17.0.7:54597->172.17.0.1:30001: read: connection reset by peer.
2016/05/18 02:41:41 transport: http2Client.notifyError got notified that the client transport was broken read tcp 172.17.0.7:54600->172.17.0.1:30001: read: connection reset by peer.
02:41:41.237 [consensus/obcpbft] execDoneSync -> INFO 038[0m Replica 4 finished execution 2, trying next
2016/05/18 02:41:41 grpc: ClientConn.resetTransport failed to create client transport: connection error: desc = "transport: write tcp 172.17.0.7:54581->172.17.0.1:30001: write: broken pipe"; Reconnecting to "172.17.0.1:30001"
2016/05/18 02:41:41 grpc: ClientConn.resetTransport failed to create client transport: connection error: desc = "transport: write tcp 172.17.0.7:54589->172.17.0.1:30001: write: broken pipe"; Reconnecting to "172.17.0.1:30001"
2016/05/18 02:41:42 grpc: ClientConn.resetTransport failed to create client transport: connection error: desc = "transport: write tcp 172.17.0.7:54595->172.17.0.1:30001: write: broken pipe"; Reconnecting to "172.17.0.1:30001"
2016/05/18 02:41:42 grpc: ClientConn.resetTransport failed to create client transport: connection error: desc = "transport: write tcp 172.17.0.7:54605->172.17.0.1:30001: write: broken pipe"; Reconnecting to "172.17.0.1:30001"
[31m02:41:43.345 [peer] chatWithPeer -> ERRO 039[0m Ending chat with peer address=172.17.0.1:30001 due to error:  Error creating handler during handleChat initiation: Error creating PeerHandler: Error creating new Peer Handler, error returned sending DISC_HELLO: Error Sending message through ChatStream: rpc error: code = 13 desc = "transport: write tcp 172.17.0.7:54593->172.17.0.1:30001: use of closed network connection"
```

Commit Level : `2b33a6f`
## NOTE:
- This issue is not seen on a 4 peer network with the above settings (`security enabled` and `pbft batch mode`)
- We can help setup the environment to recreate this issue.

[logs.zip](https://github.com/hyperledger/fabric/files/269558/logs.zip)

**Comment by tuand27613** [comment]
Does it fail consistently on a 5 peer network ?

From this one log,  it looks like the other peers crashed or something happened to the transport network. Can you attach the logs for all 5 peers ?

**Comment by ratnakar-asara** [comment]
@tuand27613 , Yes it fails consistently on a 5 peer network in vagrant environment.

Please note that the issue is not seen outside vagrant environment, i.e, with Docker setup.

logs.zip is already attached which contains logs from all peers, please let me know if you need more details.

**Comment by corecode** [comment]
Did you adjust the PBFT config for 5 peers?  I can see:
`02:39:08.001 [consensus/obcpbft] newPbftCore -> INFO 018[0m PBFT Max number of validating peers (N) = 4`

Which means that PBFT expects only 4 peers to participate.

**Comment by ratnakar-asara** [comment]
We are using a script to create the peers and run dockers using the same.
I could see docker command is missing below PBFT config.
`-e CORE_PBFT_GENERAL_N=5`

Issue is no more reproducible after modifying the script with the above env variable.

NOTE: It works fine on non-vagrant environment without having `CORE_PBFT_GENERAL_N` set.

**Comment by corecode** [comment]
I think this should be addressed with #756 and get better error messages.
